AdvancedReports.php

```diff
@@ -807,15 +824,16 @@ function parseDescription( $str )
 
 	// Returns the supplied string with any HTML entity encoded, with the exception of hyperlinks.
 	// If the $forDownload parameter is true, hyperlink tags will be stripped instead.
+	/** @psalm-pure */
 	function parseHTML( $str, $forDownload = false )
 	{
 		if ( $forDownload )
 		{
 			return preg_replace( '/<a href="[^"]*"( target="_blank")?>(.*?)<\/a>/', '$2', $str );
 		}
-		return preg_replace( '/&lt;a href="([^"]*)"( target="_blank")?&gt;(.*?)&lt;\/a&gt;/',
-		                     '<a href="$1"$2>$3</a>',
-		                     htmlspecialchars( $str, ENT_NOQUOTES ) );
+		return preg_replace( '/&lt;a href=&quot;((?(?=&quot;)|.)*)&quot;( ' .
+		                     'target=&quot;_blank&quot;)?&gt;(.*?)&lt;\/a&gt;/',
+		                     '<a href="$1"$2>$3</a>', htmlspecialchars( $str, ENT_QUOTES ) );
```

Comment:
Should this not use new escapeHTML function?

Reply:
It could but I will leave it for now.